tests/src/test/scala/whisk/core/cli/test/WskBasicUsageTests.scala

```diff
@@ -652,14 +655,43 @@ class WskBasicUsageTests
                     "key" -> JsString("web-export"),
                     "value" -> JsBoolean(true)),
                 JsObject(
+                    "key" -> JsString(origKey),
+                    "value" -> origValue),
+                JsObject(
                     "key" -> JsString("raw-http"),
                     "value" -> JsBoolean(false)),
                 JsObject(
                     "key" -> JsString("final"),
                     "value" -> JsBoolean(true)),
                 JsObject(
-                    "key" -> JsString(key),
-                    "value" -> value),
+                    "key" -> JsString(updatedKey),
+                    "value" -> updatedValue),
+                JsObject(
+                    "key" -> JsString("exec"),
+                    "value" -> JsString("nodejs:6")))
+    }
+
+    it should "ensure action update creates an action with --web flag" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "webaction"
+            val file = Some(TestUtils.getTestActionFilename("echo.js"))
+
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, file, web = Some("true"), update = true)
```

Comment:
To validate handling of existing action annotations, should the action be created first and then updated?